spacy/lang/sl/punctuation.py

```diff
@@ -0,0 +1,64 @@
+    LIST_CURRENCY, CONCAT_QUOTES
+from ..char_classes import CONCAT_QUOTES, ALPHA_LOWER, ALPHA_UPPER, ALPHA
+from ..char_classes import merge_chars
+from ..punctuation import TOKENIZER_PREFIXES as BASE_TOKENIZER_PREFIXES
+
+_prefixes = ["•", "\-", "\/", "'"] + BASE_TOKENIZER_PREFIXES
+
+_suffixes = (
+        ["—", "–", "\-", "\:", "\+", "\/", "=", "×", "\$", "•", "¯", "%"]
+        + LIST_PUNCT
+        + LIST_ELLIPSES
+        + LIST_QUOTES
+        + LIST_ICONS
+        + [
+            r"(?<=°[FfCcKk])\.",
+            r"(?<=[0-9])(?:{c})".format(c=CURRENCY),
+            r"(?<=[0-9])(?:{u})".format(u=UNITS),
+            r"(?<=[{al}{e}{p}(?:{q})])\.".format(al=ALPHA_LOWER, e=r"%²\-\+", q=CONCAT_QUOTES, p=PUNCT),
+            r"(?<=[{au}][{au}])\.".format(au=ALPHA_UPPER),
+            r"(?<=[{a}][{au}])\.".format(a=ALPHA, au=ALPHA_UPPER),
+            # split initials like J.K. Rowling
+            r"(?<=[A-Z]\.)(?:[A-Z].)",
+        ]
+)
+
+# a list of all suffixes following a hyphen that are not to be split (eg. BTC-jev)
+CONCAT_QUOTES = CONCAT_QUOTES.replace("\'", "")
+HYPHENS_PERMITTED = "((a)|(evemu)|(evskega)|(i)|(jevega)|(jevska)|(jevskimi)|(jinemu)|(oma)|(ovim)|" \
+                    "(ovski)|(e)|(evi)|(evskem)|(ih)|(jevem)|(jevske)|(jevsko)|(jini)|(ov)|(ovima)|" \
+                    "(ovskih)|(em)|(evih)|(evskemu)|(ja)|(jevemu)|(jevskega)|(ji)|(jinih)|(ova)|" \
+                    "(ovimi)|(ovskim)|(ema)|(evim)|(evski)|(je)|(jevi)|(jevskem)|(jih)|(jinim)|" \
+                    "(ove)|(ovo)|(ovskima)|(ev)|(evima)|(evskih)|(jem)|(jevih)|(jevskemu)|(jin)|" \
+                    "(jinima)|(ovega)|(ovska)|(ovskimi)|(eva)|(evimi)|(evskim)|(jema)|(jevim)|" \
+                    "(jevski)|(jina)|(jinimi)|(ovem)|(ovske)|(ovsko)|(eve)|(evo)|(evskima)|(jev)|" \
+                    "(jevima)|(jevskih)|(jine)|(jino)|(ovemu)|(ovskega)|(u)|(evega)|(evska)|" \
+                    "(evskimi)|(jeva)|(jevimi)|(jevskim)|(jinega)|(ju)|(ovi)|(ovskem)|(evem)|" \
+                    "(evske)|(evsko)|(jeve)|(jevo)|(jevskima)|(jinem)|(om)|(ovih)|(ovskemu)|" \
+                    "(ovec)|(ovca)|(ovcu)|(ovcem)|(ovcev)|(ovcema)|(ovcih)|(ovci)|(ovce)|(ovcimi)|" \
+                    "(evec)|(evca)|(evcu)|(evcem)|(evcev)|(evcema)|(evcih)|(evci)|(evce)|(evcimi)|" \
+                    "(jevec)|(jevca)|(jevcu)|(jevcem)|(jevcev)|(jevcema)|(jevcih)|(jevci)|(jevce)|" \
+                    "(jevcimi)|(ovka)|(ovke)|(ovki)|(ovko)|(ovk)|(ovkama)|(ovkah)|(ovkam)|(ovkami)|" \
+                    "(evka)|(evke)|(evki)|(evko)|(evk)|(evkama)|(evkah)|(evkam)|(evkami)|(jevka)|" \
+                    "(jevke)|(jevki)|(jevko)|(jevk)|(jevkama)|(jevkah)|(jevkam)|(jevkami)|(timi)|" \
+                    "(im)|(ima)|(a)|(imi)|(e)|(o)|(ega)|(ti)|(em)|(tih)|(emu)|(tim)|(i)|(tima)|" \
+                    "(ih)|(ta)|(te)|(to)|(tega)|(tem)|(temu))"
+
+_infixes = (
+        LIST_ELLIPSES
+        + LIST_ICONS
+        + [
```

Comment:
It would be helpful to have an explanation and preferably also a test for the first two of these rules. In the case of the first one, it might also be worth considering replacing the list of punctuation with an existing character class?

Reply:
It was just the fact that I was following the gold standard dataset that seems to split much more than the Spacys default tokenizer. Things like `naslovom»Pogovor` or `odlomek)Ciljna` would be split by » and ). But after second thought I don't think the Slovenian tokenizer should work any differently than the default tokenizer on non language specific strings. What do you think? Should we follow the dataset (and its tokenizer [Obeliks](https://github.com/clarinsi/obeliks)) or leave as much of the default tokenizer as possible?

I have cleaned the rules and commented the out for now. The only rule that would be different from the default ones then would be this one

`r"(?<=[{a}0-9])(?:{h})(?!{hp}$)".format(a=ALPHA, h=HYPHENS, hp=HYPHENS_PERMITTED),`

Is there a way to only add this one to the default infixes instead of copy pasting everything and just modifying one rule?

Reply:
If you think it's kind of a fluke in the data that those tokens are split, it's fine to leave this out from the default spacy settings. It's useful to be able to say that the default tokenizer settings for a particular language are following the guidelines from an existing corpus, but it doesn't have to be customized for every single case in the corpus if you don't think this makes sense for general spacy users.

It's not hard to add a new pattern to the default infixes, but editing or removing an existing pattern is harder, so we usually copy/paste in those cases. You can extend the infixes like this (the order can matter if there are overlapping patterns):

https://github.com/explosion/spaCy/blob/93960dc4b59510b011c12079fbba09eb8219f74e/spacy/lang/ko/punctuation.py#L1-L12

Is it possible that adding the custom rule first will cover the cases you're trying to address? (I don't actually think so, but at this point I'm not sure I understand enough to the details to be sure?)

Reply:
I've tried to add the rule before the `BASE_TOKENIZER_INFIXES` and it doesn't seem to follow it, because there is another hyphen rule after it. The rule I am trying to add just says don't split any words that end with a hyphen and the any of the suffixes from the list HYPHENS_PERMITTED. 

e.g. BTC-ju, UKC-jev, tam-ovci, SFOR-ju ...

These shouldn't be split because they contain -ju, -jev, -ovci, -ju that is in the list. Everything else containing a hyphen should be split as normal.

Reply:
Yes, I think in this case we have to copy/paste and alter the standard group (as you have done), because you're not defining new infixes but restricting the scope of one of the current infixes. However, I would change the `HYPHENS_PERMITTED` index to look like this:

`r"(?<=[{a}0-9])(?:{h})(?!{hp}$)(?=[{a}])".format(a=ALPHA, h=HYPHENS, hp=HYPHENS_PERMITTED)`

because the existing infix will match e.g. `BTC-3`.

Reply:
Yes, thanks. I've missed that one.